examples/animation/blend-trees-2d-cartesian.html

```diff
@@ -0,0 +1,220 @@
+                                    {
+                                        "name": "Idle",
+                                        "point": [0.0, 0.5]
+                                    },
+                                    {
+                                        "name": "Walk",
+                                        "point": [0.5, -0.5]
+                                    },
+                                    {
+                                        "name": "Dance",
+                                        "point": [-0.5, -0.5]
+                                    },
+                                ]
+                            },
+                        }
+                    ],
+                    "transitions": [
+                        {
+                            "from": "START",
+                            "to": "Emote"
+                        },
+                    ]
+                },
+            ],
+            "parameters": {
+                "posX": {
+                    "name": "posX",
+                    "type": "FLOAT",
+                    "value": 0
+                },
+                "posY": {
+                    "name": "posY",
+                    "type": "FLOAT",
+                    "value": 0
+                }
+            }
+        };
+
+        var canvas = document.getElementById("application-canvas");
+
+        // Create the application and start the update loop
+        var app = new pc.Application(canvas);
+
+        // Set the canvas to fill the window and automatically change resolution to be the same as the canvas size
+        app.setCanvasFillMode(pc.FILLMODE_FILL_WINDOW);
+        app.setCanvasResolution(pc.RESOLUTION_AUTO);
+
+        window.addEventListener("resize", function () {
+            app.resizeCanvas(canvas.width, canvas.height);
+        });
+
+        var miniStats = new pcx.MiniStats(app);
+
+        var entity;
+
+        // Create an Entity with a camera component
+        var cameraEntity = new pc.Entity(app);
+        cameraEntity.addComponent("camera", {
+            clearColor: new pc.Color(0.1, 0.15, 0.2)
+        });
+        cameraEntity.translateLocal(0.0, 0.75, 5.0);
+        app.root.addChild(cameraEntity);
+
+        app.scene.ambientLight = new pc.Color(0.5, 0.5, 0.5);
+
+        // Create an entity with a light component
+        var light = new pc.Entity();
+        light.addComponent("light", {
+            type: "directional",
+            color: new pc.Color(1, 1, 1),
+        });
+        light.setLocalEulerAngles(45, 30, 0);
+        app.root.addChild(light);
+
+        // Load model and animation assets
+        var model = null;
+        var idleAnim = null;
+        var danceAnim = null;
+        var walkAnim = null;
+
+        var stateGraphAsset = new pc.Asset('stateGraph', 'animstategraph', null, animStateGraph);
+        stateGraphAsset.once('load', function (asset) {
+            var stateGraph = asset.resource;
```

Comment:
perhaps update examples to use this when merged
https://github.com/playcanvas/engine/pull/2978

Reply:
Will do 👍

Reply:
I've switched these examples over to use asset loader function. I've also updated the function to handle data asset types too. Could you take a look?